core/common/test/CommonPlatformTest.kt

```diff
@@ -24,24 +24,24 @@ package kotlinx.io
 import kotlin.test.Test
 import kotlin.test.assertEquals
 
-class CommonOkioKotlinTest {
+class CommonPlatformTest {
   @Test fun sourceBuffer() {
-    val source = Buffer().writeUtf8("a")
-    val buffered = (source as RawSource).buffer()
+    val source = Buffer().also { it.writeUtf8("a") }
+    val buffered = (source as RawSource).buffered()
```

Comment:
BTW: I believe, that this cast is not needed anymore. In Okio, there is deprecated function on `BufferedSource/BufferedSink` called [`buffer()`](https://github.com/square/okio/blob/81bce1a30af244550b0324597720e4799281da7b/okio/src/jvmMain/kotlin/okio/BufferedSink.kt#L30) to get underlying `Buffer`. in `kotlinx-io` we now have only `val buffer: Buffer`

Reply:
The cast is still required when someone is trying to invoke `buffered` on a `Buffer` because it's ambiguous whether `Sink::buffered` or `Source::buffered` should be called.

I'm wondering if someone (besides tests like this) would actually want to have multiple buffered sinks or sources on top of each other.

Reply:
Yep, you got it. The only time you’d buffer a buffer is in a test.